### PR TITLE
Transport moved from deprecated to rtps [6734]

### PR DESCRIPTION
**Pull request by adolfomarver**
Just moved the transport

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/769/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/648/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/764/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/222/)

**Comment by adolfomarver** [comment]
Uncrustify & rmw_fastrtps compilation are still pending

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/801/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/676/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/796/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/250/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/802/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/677/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/797/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/251/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/803/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/678/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/798/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/252/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/807/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/682/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/802/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/256/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/810/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/685/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/805/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/259/)

File: include/fastdds/rtps/network/SenderResource.h [hunk 1]
```diff
@@ -19,14 +19,23 @@
...
+namespace fastdds{
+namespace rtps{
+
+//class ChannelResource;
+//class TransportInterface;
+
+}
+}
+}
+
```
Comment (LuisGP) [inline include/fastdds/rtps/network/SenderResource.h]: This block could be deleted.

File: include/fastrtps/transport/TCPChannelResource.h [hunk 2]
```diff
@@ -18,184 +18,24 @@
 #include <fastrtps/transport/TCPTransportDescriptor.h>
 #include <fastrtps/transport/TransportReceiverInterface.h>
 #include <fastrtps/transport/ChannelResource.h>
-#include <fastrtps/transport/tcp/RTCPMessageManager.h>
-#include <fastdds/rtps/common/Locator.h>
+//#include <fastrtps/transport/tcp/RTCPMessageManager.h>
```
Comment (LuisGP) [inline include/fastrtps/transport/TCPChannelResource.h]: Remove this kind of comment.

File: include/fastrtps/transport/TCPTransportInterface.h [hunk 3]
```diff
@@ -15,370 +15,24 @@
 #ifndef TCP_TRANSPORT_INTERFACE_H
 #define TCP_TRANSPORT_INTERFACE_H
 
+#include <string>
```
Comment (LuisGP) [inline include/fastrtps/transport/TCPTransportInterface.h]: Is this header need here?

Comment (adolfomarver) [reply include/fastrtps/transport/TCPTransportInterface.h]: Not really

Comment (LuisGP) [reply include/fastrtps/transport/TCPTransportInterface.h]: Remove it instead of commenting it out.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/919/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/765/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/914/)
* Performance Build Status

File: include/fastrtps/transport/TCPTransportInterface.h (hunk 3 above)
Comment (adolfomarver) [reply include/fastrtps/transport/TCPTransportInterface.h]: Opps

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/923/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/767/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/918/)
* Performance Build Status

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/929/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/924/)
* Performance Build Status

**Pull request merged**